modules/xz/5.4.5/patches/patch.diff

```diff
@@ -0,0 +1,5913 @@
++
++    good-1-v1-trailing-1.lz is like good-1-v1.lz but contains
++    trailing data that the decompressor must ignore.
++
++    good-1-v1-trailing-2.lz is like good-1-v1.lz but contains
++    trailing data whose first three bytes match the .lz magic bytes.
++    With lzip >= 1.20 this file results in an error unless one uses
++    the command line option --loose-trailing. lzip 1.3 to 1.19 decode
++    this file successfully by default. XZ Utils uses the old behavior
++    because it allows lzma_code() to stop at the first byte of the
++    trailing data as long as the first byte isn't 0x4C (L in US-ASCII);
++    otherwise the first 1-3 bytes that equal to the magic bytes are
++    consumed and lost in lzma_code(), and this is visible in xz too:
++
++        $ ( xz -dc ; cat ) < good-1-v1-trailing-2.lz
++        Hello
++        World!
++        Trailing garbage
++
++        $ ( xz -dc --single-stream ; cat ) < good-1-v1-trailing-2.lz
++        Hello
++        World!
++        LZITrailing garbage
++
++    good-2-v0-v1.lz contains two members of which the first is
++    version 0 and the second version 1. lzip versions 1.3 to 1.17
++    (inclusive) can decompress this.
++
++    good-2-v1-v0.lz contains two members of which the first is
++    version 1 and the second version 0. lzip versions 1.3 to 1.17
++    (inclusive) can decompress this.
++
++    good-2-v1-v1.lz contains two version 1 members. lzip versions 1.3
++    and newer can decompress this.
++
++
++4.2. Unsupported Files
++
++    unsupported-1-v234.lz is like good-1-v1.lz except the version
++    field has been set to 234 (0xEA) which, as of writing, isn't
++    defined or supported by any .lz implementation.
++
++
++4.3. Bad Files
++
++    bad-1-v1-magic-1.lz is like good-1-v1.lz but the first magic byte
++    is wrong.
++
++    bad-1-v1-magic-2.lz is like good-1-v1.lz but the last (fourth)
++    magic byte is wrong.
++
++    bad-1-v1-dict-1.lz has too low value in the dictionary size field.
++
++    bad-1-v1-dict-2.lz has too high value in the dictionary size field.
++
++    bad-1-v1-crc32.lz has wrong CRC32 value.
++
++    bad-1-v0-uncomp-size.lz is version 0 format with incorrect value
++    in the uncompressed size field.
++
++    bad-1-v1-uncomp-size.lz is version 1 format with incorrect value
++    in the uncompressed size field.
++
++    bad-1-v1-member-size.lz has incorrect value in the member size
++    field.
++
++    bad-1-v1-trailing-magic.lz has the four .lz magic bytes as trailing
++    data. This should be detected as a truncated file and thus result
++    in an error. That is, the last four bytes of the file should not be
++    ignored as trailing garbage. lzip >= 1.18 matches this behavior
++    while older versions ignore the last four bytes and don't indicate
++    an error.
++
+diff --git a/files/bad-0-backward_size.xz b/files/bad-0-backward_size.xz
+new file mode 100644
+index 0000000..2b46fa9
+Binary files /dev/null and b/files/bad-0-backward_size.xz differ
```

Comment:
just to double-check, are these segments intended or useful at all?

Reply:
the CI failure suggests we do need to fix these.

Reply:
Yes, they are data used by the tests. They had to be moved to root level which is why they are in the diff, there might be a better way to do this though. Not sure why the patch didn't work properly, looking into it now

Reply:
Was a dodgy patch file, not sure why but Sourcetree doesn't patch file moves properly. Used git diff --cached > patch.diff and seems fine now